fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -612,6 +612,309 @@ void partitionedJoinedIndexTest() {
+
+            // partition 1 covers the past 29 days till yesterday
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 1,
+                    Instant.now().minus(29, ChronoUnit.DAYS).toEpochMilli(),
+                    Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+
+            // this gets indexed into partition 0
+            recordStore.saveRecord(createComplexDocument(6666L, ENGINEER_JOKE, 1, Instant.now().minus(40, ChronoUnit.DAYS).toEpochMilli()));
+
+            // this gets indexed into partition 1
+            recordStore.saveRecord(createComplexDocument(7777L, ENGINEER_JOKE, 1, Instant.now().minus(20, ChronoUnit.DAYS).toEpochMilli()));
+            recordStore.saveRecord(createComplexDocument(8888L, ENGINEER_JOKE, 1, Instant.now().minus(20, ChronoUnit.DAYS).toEpochMilli()));
+
+            // query will return results from both partitions
+            assertIndexEntryPrimaryKeyTuples(Set.of(Tuple.from(1, 7777L), Tuple.from(1, 6666L), Tuple.from(1, 8888L)),
+                    recordStore.scanIndex(COMPLEX_PARTITIONED, groupedTextSearch(COMPLEX_PARTITIONED, "text:propose", 1), null, ScanProperties.FORWARD_SCAN));
+            assertEquals(3, getCounter(context, FDBStoreTimer.Counts.LOAD_SCAN_ENTRY).getCount());
+
+            commit(context);
+        }
+    }
+
+    /**
+     * test LIMIT spanning partitions.
+     */
+    @Test
+    void testPartitionedLimit() {
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+
+            // partition 0 covers 30 days beginning 60 days ago
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 0,
+                    Instant.now().minus(60, ChronoUnit.DAYS).toEpochMilli(),
+                    Instant.now().minus(30, ChronoUnit.DAYS).toEpochMilli());
+
+            // partition 1 covers the past 29 days till yesterday
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 1,
+                    Instant.now().minus(29, ChronoUnit.DAYS).toEpochMilli(),
+                    Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+
+            // 30 records in partition 0
+            for (int i = 0; i < 30; i++) {
+                recordStore.saveRecord(createComplexDocument(1111L + i, ENGINEER_JOKE, 1, Instant.now().minus(40, ChronoUnit.DAYS).toEpochMilli()));
+            }
+
+            // 30 records in partition 1
+            for (int i = 0; i < 30; i++) {
+                recordStore.saveRecord(createComplexDocument(2222L + i, ENGINEER_JOKE, 1, Instant.now().minus(20, ChronoUnit.DAYS).toEpochMilli()));
+            }
+
+            // get 50 records (30 from partition 1, and 20 from partition 0)
+            assertEquals(50, recordStore.scanIndex(COMPLEX_PARTITIONED, groupedTextSearch(COMPLEX_PARTITIONED, "text:propose", 1), null, ExecuteProperties.newBuilder().setReturnedRowLimit(50).build().asScanProperties(false))
```

Comment:
since each partition only has 30 docs, then getting 50 back necessarily means they came from both p0 and p1. Unless you mean that the 50 break down to 30 from p1 and 20 from p0.